frontend/kesaseteli/employer/src/components/header/Header.tsx

```diff
@@ -36,17 +35,11 @@ const Header: React.FC = () => {
     [router, asPath]
   );
 
-  const onError = useErrorHandler(false);
-
   const login = useLogin();
   const userQuery = useUserQuery();
-  const logoutQuery = useLogoutQuery();
-  const logout = React.useCallback(
-    () => logoutQuery.mutate({}, { onError }),
-    [logoutQuery, onError]
-  );
+  const logout = useLogout();
```

Comment:
The problem is that the useLogoutQuery cleans cache after success and now it does not do it:
If you check the `useLogoutQuery.tsx` there is success callback:
```
      onSuccess: () => {
        void queryClient.removeQueries();
        clearLocalStorage('application');
        void goToPage('/login?logout=true');
      },
```
goToPage call is no longer needed but we still need to clear local storage and remove react-query cache after succesfull logout. Perhaps you can do it on the `login.tsx` page on `useEffect`when `logout` value is true which means that we have just logged out succesfully.